core/src/main/java/nl/nn/adapterframework/configuration/digester/ValidateAttributeRule.java

```diff
@@ -141,74 +111,42 @@ private <E extends Enum<E>> E parseAsEnum(Class<?> enumClass, String fieldName,
-				}
-				// if the default value is null, then it can mean that the real default value is determined in configure(),
-				// so we cannot assume setting it to "" has no effect
-			} catch (NumberFormatException e) {
-				addLocalWarning("attribute ["+ name+"] with value ["+value+"] cannot be converted to a number: "+e.getMessage());
-			} catch (Throwable t) {
-				addLocalWarning("is unable to parse attribute ["+name+"] value ["+value+"] to method ["+rm.getName()+"] with type ["+rm.getReturnType()+"]");
-				log.warn("Error on getting default for object [" + getObjectName() + "] with method [" + rm.getName() + "] attribute ["+name+"] value ["+value+"]", t);
-			}
-		} else {
-			//No readMethod, thus we cannot check the default value. We can however check if we can parse the value
-			if (value.length()==0) {
-				// no need to check if we can parse an empty string, but no warning either, as empty string might be an appropriate value
-				return;
+	private void checkTypeCompatibility(Method readMethod, String name, String value, Map<String, String> attrs) {
+		try {
+			Object bean = getBean();
+			Object defaultValue = readMethod.invoke(bean);
+			if (bean instanceof HasSpecialDefaultValues) {
+				defaultValue = ((HasSpecialDefaultValues)bean).getSpecialDefaultValue(name, defaultValue, attrs);
 			}
-
-			//If it's a number (int/long) try to parse it, else BeanUtils will call the method with 0.
-			try {
-				Class<?> setterArgumentClass = pd.getWriteMethod().getParameters()[0].getType();
-
-				switch (setterArgumentClass.getTypeName()) {
-				case "int":
-				case "java.lang.Integer":
-					Integer.parseInt(value);
-					break;
-				case "long":
-					Long.parseLong(value);
-					break;
-				}
-			} catch(NumberFormatException e) {
-				addLocalWarning("attribute ["+name+"] with value ["+value+"] cannot be converted to a number");
-			} catch(Exception e) {
-				log.debug("unable to get the first setter parameter of attribute["+name+"] writeMethod ["+pd.getWriteMethod()+"]", e);
+			if (equals(defaultValue, value)) {
+				addSuppressableWarning("attribute ["+name+"] already has a default value ["+value+"]", SuppressKeys.DEFAULT_VALUE_SUPPRESS_KEY);
 			}
+			// if the default value is null, then it can mean that the real default value is determined in configure(),
+			// so we cannot assume setting it to "" has no effect
+		} catch (Exception e) {
+			addLocalWarning("is unable to parse attribute ["+name+"] value ["+value+"] to method ["+readMethod.getName()+"] with type ["+readMethod.getReturnType()+"]");
+			log.warn("Error on getting default for object [" + getObjectName() + "] with method [" + readMethod.getName() + "] attribute ["+name+"] value ["+value+"]", e);
 		}
 	}
 
-	/**
-	 * Fancy equals that type-checks one value against another.
-	 */
+	/** Fancy equals that type-checks the attribute value against the defaultValue. */
 	private boolean equals(Object defaultValue, String value) {
```

Comment:
1. Kan dit de `ClassUtils.convertToType()` method gebruiken om de string om te zetten, en dan te vergelijken?
2. Wordt de case van `Enum` default-value ergens afgehandeld buiten deze method? Ik kan niet zien of dat mogelijk is of nodig is.